presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/api/PackageAPI.java

```diff
@@ -36,87 +36,86 @@
 import javax.ws.rs.Path;
 import javax.ws.rs.Produces;
 import javax.ws.rs.QueryParam;
-import javax.ws.rs.core.MediaType;
 import javax.ws.rs.core.Response;
 
 import java.net.MalformedURLException;
 import java.net.URL;
 
-import static com.google.common.io.Files.getFileExtension;
-import static javax.ws.rs.client.Entity.text;
+import static com.teradata.prestomanager.agent.CommandExecutor.executeCommand;
+import static javax.ws.rs.core.MediaType.TEXT_PLAIN;
 import static javax.ws.rs.core.Response.Status.ACCEPTED;
 import static javax.ws.rs.core.Response.Status.BAD_REQUEST;
+import static javax.ws.rs.core.Response.Status.CONFLICT;
+import static javax.ws.rs.core.Response.Status.INTERNAL_SERVER_ERROR;
 
 @Path("/package")
 @Api(description = "the package API")
 @Singleton
+// TODO: Add configuration to get package type from PM config file
 public final class PackageAPI
 {
     private static final Logger LOGGER = LogManager.getLogger(PackageAPI.class);
-    private static PackageType installedPackageType;
 
     @PUT
-    @Consumes({MediaType.TEXT_PLAIN})
-    @Produces({MediaType.TEXT_PLAIN})
+    @Consumes({TEXT_PLAIN})
+    @Produces({TEXT_PLAIN})
     @ApiOperation(value = "Install presto using rpm or tarball")
     @ApiResponses(value = {
             @ApiResponse(code = 202, message = "Acknowledged request"),
-            @ApiResponse(code = 400, message = "Bad package format or Invalid url")
+            @ApiResponse(code = 409, message = "Presto is running"),
+            @ApiResponse(code = 400, message = "Invalid url")
     })
     public synchronized Response install(@NotNull @ApiParam("Url to fetch package") String locationToFetchPackage,
```

Comment:
Using `@NotNull` here doesn't really do anything.

The methods of this class shouldn't be called by hand, so it's not particularly useful as documentation (and if we want it for that, we should have many more such annotations). And Jersey doesn't care about the `NotNull` annotation; if it passes in `null`, we need to check anyway.

Speaking of which, I believe that it will use an empty string, not `null`, when given no body, and we could check for that. "Expected URL in body" is more useful as a response than "Invalid URL".